### ISSUE TITLE
@grpc/grpc-js documentation discovery needs streamlining

### ISSUE THREAD

**Issue by tlhunter** [grpc/grpc-node#1540]
### Is your feature request related to a problem? Please describe.
I'm unable to easily documentation for `@grpc/grpc-js`.

### Describe the solution you'd like
I think the following would help me and any other engineer trying to learn to use this project:

- The [@grpc/grpc-js npm page](https://www.npmjs.com/package/@grpc/grpc-js) should link to a documentation page
- The [@grpc/grpc-js github page](https://github.com/grpc/grpc-node/tree/master/packages/grpc-js) should link to a documentation page
- The [grpc.io Node.js page](https://grpc.io/docs/languages/node/) should be updated to contain docs for `@grpc/grpc-js`, instead of `grpc`
- The [Basics Tutorial](https://grpc.io/docs/languages/node/basics/) page should contain up to date `@grpc/grpc-js` examples
- The [Node.js gRPC Library](https://grpc.github.io/grpc/node/) has a broken API Documentation link. Currently it just links to itself. It's also counter-intuitive that the top navigation is the most important and that the side navigation and body content on the landing page isn't useful.

### Describe alternatives you've considered
Lots of google searches. Eventually I had to dig into the test suite to find out how to use certain methods.

### Additional context
A Node.js developer looking to use gRPC will first land on either the GitHub page or the npm package page. The flow from getting to the package page to the underlying docs page is very painful. Once the developer does arrive at the docs page they're greeted with docs for the deprecated `grpc` module. For example, [there is no explanation](https://grpc.github.io/grpc/node/grpc.Server.html#bind__anchor) that `Server#bind` will not work with `@grpc/grpc-js`.

Modifying the docs to contain more real-world examples would be nice. Instead of documenting each method, document sets of methods and provide a full working consumer and producer example (I'll gladly donate the examples I'm building for my upcoming book Distributed Node.js).

**Comment by murgatroid99** [comment]
I mostly agree with what you're saying here, but in your last bullet point, that link goes to a page that is generated from the README file, so there isn't a good way to both have the documentation link in the README (for the first two points), and not have that page link to itself.

Also, your suggestion about rearranging the documentation doesn't seem easy. That is generated API documentation from code comments, so there are limitations on how it can be presented. There may be a place for your suggested grouped documentation, but I don't know where.

**Comment by celesteking** [comment]
Docs need to be rewritten in human terms, not in API terms. You need to bring into team someone from JS world.
https://grpc.github.io/grpc/node/grpc.Client.html  -- this is C thinking, this is not how a JS person would write docs.
This is how JS docs are written https://www.npmjs.com/package/generic-pool 

JS is for stupid people, so make the docs so.

**Comment by tlhunter** [comment]
> doesn't seem easy
> @murgatroid99

Absolutely. The gRPC package docs needs a complete overhaul. While I don't agree with a lot of what @celesteking is saying, I do agree that automatically generated documentation is never going to be ideal documentation.
